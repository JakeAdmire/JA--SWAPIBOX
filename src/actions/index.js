export const addFavorite = (film) => ({
  type: 'ADD_FAVORITE',
  film
})

export const removeFavorite = (film) => ({
  type: 'REMOVE_FAVORITE',
  film
})

export const setCategory = (event) => ({
  type: 'SET_CATEGORY',
  event
})