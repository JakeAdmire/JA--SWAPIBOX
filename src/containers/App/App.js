import React, { Component } from 'react';
import { connect } from 'react-redux';
import { chooseFilm } from '../../actions';
import { Route } from 'react-router-dom';

import { fetchHappens } from '../../helpers/fetch';
import { Crawl } from '../../components/Crawl/Crawl';
import Header from '../../components/Header/Header';
import Main from '../../components/Main/Main';

export class App extends Component {
  constructor(props) {
    super(props);
    this.state = { };
  }

  componentDidMount() {
    fetchHappens('https://swapi.co/api/films/')
      .then(films => this.props.chooseFilm(films.results))
      .catch(Error => this.setState({error: Error.message}) )
  }

  render() {
    const { crawl, title, date } = this.props.film;
    const crawlContent = { crawl, title, date };
    return (
      <div className="App">
        <Header />
        <Crawl {...crawlContent}/>
        { this.props.category && <Main /> }
      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  film: state.film,
  category: state.category
})

export const mapDispatchToProps = (dispatch) => ({
  chooseFilm: (films) => dispatch(chooseFilm(films))
})

export default connect(mapStateToProps, mapDispatchToProps)(App);