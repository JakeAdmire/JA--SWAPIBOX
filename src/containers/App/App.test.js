import { enzyme, shallow } from 'enzyme';
import React from 'react'
import { App, mapStateToProps, mapDispatchToProps } from './App';
import { chooseFilm } from '../../actions';

describe('App', () => {

  let wrapper;

  let mockProps = {
    film: {
      crawl: 'Once upon a time', 
      title: 'Movie1', 
      date: '3-28-2019'
    },
    faves: [],
  };

  wrapper = shallow(
    <App {...mockProps}/>
  )

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  }) 

})

describe('mapStateToProps', () => {

  it('should return a film object and faves array as props', () => {
    const mockState = {
      film: {},
      faves: []
    }

    const results = mapStateToProps(mockState)
    expect(results).toEqual(mockState)
  })

})

describe('mapDispatchToProps', () => {

  it('should provide a method to dispatch an action creator', () => {
    const mockFilms = [{}, {}];
    const mockDispatch = jest.fn();
    const actionToDispatch = chooseFilm(mockFilms);
    const mappedProps = mapDispatchToProps(mockDispatch);

    mappedProps.chooseFilm(mockFilms);

    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  })

})