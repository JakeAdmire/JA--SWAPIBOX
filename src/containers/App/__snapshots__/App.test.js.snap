// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    faves={Array []}
    film={
      Object {
        "crawl": "Once upon a time",
        "date": "3-28-2019",
        "title": "Movie1",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Connect(Header) />,
        <Crawl
          crawl="Once upon a time"
          date="3-28-2019"
          title="Movie1"
        />,
        <Route
          path="/:id"
          render={[Function]}
        />,
      ],
      "className": "App",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "crawl": "Once upon a time",
          "date": "3-28-2019",
          "title": "Movie1",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "path": "/:id",
          "render": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Connect(Header) />,
          <Crawl
            crawl="Once upon a time"
            date="3-28-2019"
            title="Movie1"
          />,
          <Route
            path="/:id"
            render={[Function]}
          />,
        ],
        "className": "App",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "crawl": "Once upon a time",
            "date": "3-28-2019",
            "title": "Movie1",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "path": "/:id",
            "render": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
